Bugfix/eventbridge/event bus desciption not asigned on create

maxhoheiser opened the pull request:
<!-- Please refer to the contribution guidelines before raising a PR: https://github.com/localstack/localstack/blob/master/docs/CONTRIBUTING.md -->

<!-- Why am I raising this PR? Add context such as related issues, PRs, or documentation. -->
## Motivation
Event Bridge Buses can have a description, this description is returned on create and on describe event buses.

This PR fixes the issue raised in this github bug report: https://github.com/localstack/localstack/issues/12065

<!-- What changes does this PR make? How does LocalStack behave differently now? -->
## Changes
Correctly store and handle the description of an event bridge bus on create.

## Testing
Add test for creating and describing event bridge bus with description

bentsku commented on localstack-core/localstack/services/events/event_bus.py:
```diff
@@ -45,6 +46,7 @@ def create_event_bus_service(
                 region,
                 account_id,
                 event_source_name,
+                description,
```
as a note, I think it would be good to use keyword arguments here, it's much cleaner and readable (named parameters)

bentsku commented on localstack-core/localstack/services/events/provider.py:
```diff
@@ -963,6 +963,8 @@ def create_event_bus(
         response = CreateEventBusResponse(
             EventBusArn=event_bus_service.arn,
         )
+        if description := event_bus_service.event_bus.description:
+            response["Description"] = description
```
in the case of the data being restored from persisted state, I'm not sure it's fully possible but the `description` field could be non-existent. Until the next minor release, it could be good to use `getattr(event_bus_service.event_bus, "description", None)` to be on the safe side. 

one example of a PR who removed such things: #11809 (and the PR implementing one: https://github.com/localstack/localstack/pull/11358, see this comment: https://github.com/localstack/localstack/pull/11358/files#r1717458334)

maxhoheiser replied:
fixed

bentsku reviewed (commented):
Neat fix! This is great, and good thinking with the parametrization of the test!

I have a nit about accessing the model attribute like that, as we can restore persisted state that might not have the field yet. Using `getattr` can avoid weird errors popping up and not break existing users' setup. 

Once this is addressed, I think we merge the PR. Nice fix! 🚀

bentsku commented on localstack-core/localstack/services/events/provider.py:
```diff
@@ -1945,6 +1949,8 @@ def _event_bus_to_api_type_event_bus(self, event_bus: EventBus) -> ApiTypeEventB
             "Name": event_bus.name,
             "Arn": event_bus.arn,
         }
+        if event_bus.description:
+            event_bus_api_type["Description"] = event_bus.description
```
There's still the access here, sorry I didn't put a comment 😅

bentsku reviewed (approved):
LGTM! A small comment for the `getattr` usage could be good, with a TODO, to know to revisit and remove once we switch for example major versions, but a quick search will do too! Thanks for addressing the comments!

maxhoheiser commented:
> LGTM! A small comment for the `getattr` usage could be good, with a TODO, to know to revisit and remove once we switch for example major versions, but a quick search will do too! Thanks for addressing the comments!

I will track it as a todo in the servicepage.
